File: src/main/java/org/lanternpowered/porygen/util/RegexPatterns.java
/*
 * This file is part of Porygen, licensed under the MIT License (MIT).
 *
 * Copyright (c) LanternPowered <https://www.lanternpowered.org>
 * Copyright (c) contributors
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the Software), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package org.lanternpowered.porygen.util;

import java.util.regex.Pattern;

public final class RegexPatterns {

    /**
     * A {@link Pattern} that matches double values.
     *
     * <a href="https://www.regular-expressions.info/floatingpoint.html">
     *     Matching Floating Point Numbers with a Regular Expression</a>
     */
    public static final Pattern DOUBLE_PATTERN = Pattern.compile("[-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?");

    /**
     * A {@link Pattern} that matches a double range. The syntax is: a..b
     * <p>For example: 3.32..9.11, min is 3.32 and max is 9.11
     */
    public static final Pattern DOUBLE_RANGE_PATTERN = Pattern.compile("(" + DOUBLE_PATTERN + ")\\s*\\.\\.\\s*(" + DOUBLE_PATTERN + ")");
}

File: src/main/kotlin/org/lanternpowered/porygen/util/RegexPatterns.kt
/*
 * This file is part of Porygen, licensed under the MIT License (MIT).
 *
 * Copyright (c) LanternPowered <https://www.lanternpowered.org>
 * Copyright (c) contributors
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the Software), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */
package org.lanternpowered.porygen.util

import java.util.regex.Pattern

object RegexPatterns {

    /**
     * A [Pattern] that matches double values.
     *
     * [Matching Floating Point Numbers with a Regular Expression](https://www.regular-expressions.info/floatingpoint.html)
     */
    val DOUBLE_PATTERN: Pattern = Pattern.compile("[-+]?[0-9]*\\.?[0-9]+(?:[eE][-+]?[0-9]+)?")

    /**
     * A [Pattern] that matches a double range. The syntax is: a..b
     *
     * For example: 3.32..9.11, min is 3.32 and max is 9.11
     */
    val DOUBLE_RANGE_PATTERN: Pattern = Pattern.compile("($DOUBLE_PATTERN)\\s*\\.\\.\\s*($DOUBLE_PATTERN)")
}
